routing/pathfind_test.go

```diff
@@ -616,6 +638,163 @@ func TestKShortestPathFinding(t *testing.T) {
 	assertExpectedPath(t, paths[1], "roasbeef", "satoshi", "luoji")
 }
 
+func TestNewRoute(t *testing.T) {
```

Comment:
the `newRoute` changes should be removed, since they were separated into a new PR.

Reply:
This PR is based upon the other PR, because I need the TestNewRoute function. Those changes are shown here to because the base of this PR is master. But really, all commits in this PR are stacked on top of the newRoute bugfix. Please let me know how to do this better.

Reply:
👍